ros2_batch_job/__main__.py

```diff
@@ -150,13 +150,12 @@ def main(sysargv=None):
             'qt_dotgraph',
         ]
 
-    # TODO(wjwwood): remove this when a better solution is found, as
-    #   this is just a work around for https://github.com/ros2/build_cop/issues/161
-    # If on Windows, kill any still running `colcon` processes to avoid
-    # problems when trying to delete files from pip or the workspace during
-    # this job.
+    # If on Windows, run the subprocess reaper to kill any still running job
+    # processes to avoid problems when trying to delete files from pip or the
+    # workspace during this job.
     if sys.platform == 'win32':
-        os.system('taskkill /f /im colcon.exe')
+        subprocess_reaper_path = which('subprocess_reaper.py')
+        subprocess.Popen([sys.executable, subprocess_reaper_path, str(os.getpid())])
```

Comment:
I think the process should be detached to prevent it being killed before it can do its job once this script exits, like here: https://github.com/ros2/ros2cli/blob/c00dec0a72c049d3a4a8a80f1324ea24dc8373c6/ros2cli/ros2cli/node/daemon.py#L51-L67

Also piping the output of the script into a file would be good for debugging: like https://github.com/ros-infrastructure/ros_buildfarm/blob/a322b0793986fa3b83feb794a0daca8c781fc4c2/ros_buildfarm/templates/ci/ci_job.xml.em#L151

Reply:
> I think the process should be detached to prevent it being killed before it can do its job once this script exits, like here: [ros2/ros2cli:ros2cli/ros2cli/node/daemon.py@`c00dec0`#L51-L67](https://github.com/ros2/ros2cli/blob/c00dec0a72c049d3a4a8a80f1324ea24dc8373c6/ros2cli/ros2cli/node/daemon.py#L51-L67)

The subprocess_reaper script [ignores SIGTERM](https://github.com/ros-infrastructure/ros_buildfarm/blob/a322b0793986fa3b83feb794a0daca8c781fc4c2/scripts/subprocess_reaper.py#L172) but looking at the log file it seems that Windows is sending SIGINT or similar instead causing a `KeyboardInterrupt`. I would like to keep it attached but to do so I need to make it un-^C-able on Windows. 


> Also piping the output of the script into a file would be good for debugging: like [ros-infrastructure/ros_buildfarm:ros_buildfarm/templates/ci/ci_job.xml.em@`a322b07`#L151](https://github.com/ros-infrastructure/ros_buildfarm/blob/a322b0793986fa3b83feb794a0daca8c781fc4c2/ros_buildfarm/templates/ci/ci_job.xml.em#L151)

Done.